server/util.go

```diff
@@ -323,3 +324,26 @@ func dumpTextRow(buffer []byte, columns []*ColumnInfo, row types.Row) ([]byte, e
 	}
 	return buffer, nil
 }
+
+const (
+	expFormatBig   = 1e16
+	expFormatSmall = 1e-15
+)
+
+func appendFormatFloat(in []byte, fVal float64, prec, bitSize int) []byte {
+	absVal := math.Abs(fVal)
+	var out []byte
+	if prec == -1 && absVal >= expFormatBig || (absVal != 0 && absVal < expFormatSmall) {
```

Comment:
How about the `prec` less than -1?
Please define a constant as -1. We use it in many places.